src/main/java/frc/robot/subsystems/SwerveSubsystem.java

```diff
@@ -223,10 +249,76 @@ public Command followTrajectory(String name, PathPlannerTrajectory trajectory){
         ).beforeStarting(() -> trajectoryLog.append("Following trajectory " + name)
         ).andThen(() -> trajectoryLog.append("Trajectory " + name +  " Ended"));
     }
-    
+
+    public Command alignToTarget(NodePosition nodePosition){
+        Integer id = Photonvision.getAprilTagID();
+        double xOffset = 0; //Range offset //TODO: Add constants to calculate 
+        double yOffset = 0; //Left/Right offset //TODO: Add TOF sensor method to calculate
+
+        if(id != -1 && Robot.isRed ? id < 4 : id > 5){
+            //Get pose of node to be targeted
+            Pose2d desiredPose = new Pose2d(
+                nodePosition.getTranslation(id),
+                null
+            );
+            //Apply offsets and add a rotation to get desired robot pose
+            desiredPose = new Pose2d(
+                desiredPose.getX() + xOffset, 
+                desiredPose.getY() + yOffset, 
+                Robot.isRed ? Rotation2d.fromDegrees(0) : Rotation2d.fromDegrees(180)
+            );
```

Comment:
Using Translation2d rather than individual doubles can make things a bit more concise. 

```suggestion
        Translation2d offset = new Translation2d(0,0); // TODO: calculate offsets

        if(id != -1 && Robot.isRed ? id <= 4 : id > 4){
            //Get pose of node to be targeted
            Pose2d desiredPose = new Pose2d(
                nodePosition.getTranslation(id).plus(offset),
                Robot.isRed ? Rotation2d.fromDegrees(0) : Rotation2d.fromDegrees(180)
            );
```